wgsl/index.bs

```diff
@@ -1117,10 +1165,19 @@ A structure member type must be one of:
 * a matrix type
 * an [=atomic type|atomic=] type
 * an array type
-* a [=structure=] type
+* a [=structure=] type that is not a [=buffer-root=]
 
 Note: That is, each member type must be a [=plain type=].
 
+A structure type is a <dfn>buffer-root</dfn> if it is used as the [=store type=]
```

Comment:
This seems a bit strange, it's backwards. It says, the type depends on how it's used.
I guess one could describe type inference in a similar way, so it's not too exceptional

Reply:
An alternative would be that each struct definition could potentially expand into _two_ SPIR-V constructs: one block and one non-block. Then it would be fine to use the same struct as both buffer-root and non-buffer-root.

It depends on how much you want to abstract away SPIR-V's "block" concept.

Reply:
Another alternative is to keep `[[block]]`.

Thinking about this more, this change carries a **SAaaD** hazard (Spooky Action at a Distance): if you have a large module, and you decide to add a uniform binding for a particular struct, you may get an error in potentially a totally different place (where this structure was used before, but now is not valid).

Reply:
Agree for the PR as is. If the WGSL struct could expand into two SPIR-V structs I think it would avoid SAaaD?

Reply:
if SPIR-V has to split the structs, you don't need to define the whole "buffer-root" concept. While in general I'd be very wary against splitting types, the main complexity that comes with it is dealing with all the composite dependent types. In case of `[[block]]` there are not dependent types by definition, so splitting could be done without much blood lost.